core/chains/evm/client/rpc_client.go

```diff
@@ -173,12 +176,15 @@ func NewRPCClient(
 
 func (r *RpcClient) SubscribeToHeads(ctx context.Context) (<-chan *evmtypes.Head, commontypes.Subscription, error) {
 	channel := make(chan *evmtypes.Head)
-	sub, err := r.Subscribe(ctx, channel)
+	sub, err := r.subscribe(ctx, channel, rpcSubscriptionMethodNewHeads)
 	return channel, sub, err
 }
 
 func (r *RpcClient) SubscribeToFinalizedHeads(_ context.Context) (<-chan *evmtypes.Head, commontypes.Subscription, error) {
 	interval := r.cfg.FinalizedBlockPollInterval()
+	if interval == 0 {
```

Comment:
This seems weird.
Why do we then allow this config to be set to 0?

Reply:
It shouldn't be 0, this is really just a sanity check.